### DISCUSSION TITLE
Is there a Rust usage/how-to guide ?

### DISCUSSION THREAD

**Discussion by avhz** [EnzymeAD/Enzyme#1756] [Q&A] [answered]
Hi :) 

Just wondering if there is a guide to using Enzyme in a Rust library ? 

Thanks for the cool work !

**Comment by ZuseZ4** [comment, marked as answer]
Hi, happy you are interested in the Rust side!
I have written some early documentation in 
http://enzyme.mit.edu/rust, especially
https://enzyme.mit.edu/index.fcgi/rust/usage/usage.html
The rust repo itself is under EnzymeAD/rust, this here is for Enzyme Core

**Reply by wsmoses** [reply]
Edit: This repo is not just for C/C++ enzyme, but Enzyme core (e.g. all the inner workings of the Enzyme AD system). The Rust frontend however, is in the above repo

**Reply by avhz** [reply]
Is it possible to use Enzyme in a Rust project without using the forked Rust compiler at https://github.com/EnzymeAD/rust ? 

I noticed an old archived crate on crates.io, but can't seem to see anything else. 

Cheers for the advice!

**Reply by ZuseZ4** [reply]
Not yet, we are currently at the upstreaming step.
https://github.com/rust-lang/rust/issues/124509
Once I got to it, I can ping you.

**Reply by avhz** [reply]
That would be awesome, cheers!
Thanks for pointing out the tracking issue too, hadn't noticed it. 

Would be very cool to see Enzyme support in nightly, fingers crossed.
